### DISCUSSION TITLE
Make assigning the same list of VLANs to multiple interfaces easier

### DISCUSSION THREAD

**Discussion by billcouper81** [netbox-community/netbox#16520] [Ideas]
Virtualization hosts in a pool/cluster generally require the same list of VLANs across multiple interfaces per server. Manually applying the same list of VLANs to large numbers of interfaces is time consuming and prone to human error.

It is currently not possible to easily duplicate a list of VLANs from one interface to another. At least, not within the GUI (that I can see - PLEASE tell me I am missing something here!)

At a minimum please add a 'copy vlan list' button when editing an interface. Perhaps it could simply copy a comma-separated list of VLAN numbers to the clipboard. Then, allow a comma-separated list of VLANs to be pasted into the list when editing an interface.

What I'd really like to see is to manage VLANs at a virtualization cluster level. Perhaps we could create a list of VLANs that are assigned to a virtualization cluster, and the option to include/exclude cluster host interfaces from this VLAN list assignment.

**Comment by einhirn** [comment]
Currently you'd have to fiddle around with API to 

- extract the existing list of VLAN-object-ids on a "template" interface
- find the VLAN-object-id for the VLAN to add
- find the object IDs for all interfaces you want to replace the list of vlans of
- write JSON to "PATCH" on `/api/dcim/interfaces` and run the query...
